Marlin/src/gcode/feature/pause/M600.cpp

```diff
@@ -43,11 +49,35 @@
  *  L[distance] - Extrude distance for insertion (positive value) (manual reload)
  *  B[count]    - Number of times to beep, -1 for indefinite (if equipped with a buzzer)
  *
+ * Spool_Swap : Tool change after filament runout
+ *
+ *  T[int]: 0=disable : 1/2/3 Enable the max number of swapping
+ *
  *  Default values are used for omitted arguments.
  *
  */
 void GcodeSuite::M600() {
-
+	
+  #if ENABLED(ADVANCED_PAUSE_SPOOL_SWAP) 
+   
+	const int swap_spool_tmp = parser.intval('T',(FIL_RUNOUT_SENSORS));             
+    
+	// M600 T0/T1/2/3/4 Enable/Disable     
+	if (swap_spool_tmp<FIL_RUNOUT_SENSORS){
+	  if (swap_spool_tmp>0){
+	    // Activate or deactivate if next extruder available
+	    swap_spool_enabled = true ;
+		//Apply max number of swapping 
+		swap_spool_stop = swap_spool_tmp;
+	  }
+	  // Deactivate in all case if no next extruders available
+	  else swap_spool_enabled=false; 
+	  // Applying number or next extruders
+	  
+	  return ;  
+    }      
+  #endif
+	
```

Comment:
M600 T0 = OFF
M600 T1  = ON and only '1' swap allowed
M600 T2  = ON and only '2' swaps allowed 
But , if not sensors available , or no extruders available , spool swapping = OFF